backend/analysis/views.py

```diff
@@ -0,0 +1,137 @@
+from django.core.exceptions import BadRequest
+from drf_spectacular.types import OpenApiTypes
+from drf_spectacular.utils import extend_schema, OpenApiResponse, OpenApiExample, inline_serializer
+from rest_framework import status
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
+from rest_framework.views import APIView
+from rest_framework import serializers
+
+from analysis.serializers import WorkedHoursAnalysisSerializer, StudentOnTourAnalysisSerializer
+from base.models import StudentOnTour, RemarkAtBuilding
+from base.permissions import IsAdmin, IsSuperStudent
+from util.request_response_util import (
+    get_success,
+    get_filter_object,
+    filter_instances,
+    bad_request_custom_error_message,
+    not_found,
+    param_docs,
+)
+
+
+class WorkedHoursAnalysis(APIView):
+    permission_classes = [IsAuthenticated, IsAdmin | IsSuperStudent]
+    serializer_class = WorkedHoursAnalysisSerializer
+
+    @extend_schema(
+        description="Get all worked hours for each student for a certain period",
+        parameters=param_docs(
+            {
+                "start-date": ("Filter by start-date", True, OpenApiTypes.DATE),
+                "end-date": ("Filter by end-date", True, OpenApiTypes.DATE),
+                "region_id": ("Filter by region id", False, OpenApiTypes.INT),
+            }
+        ),
+        responses={
+            200: OpenApiResponse(
+                description="All worked hours for each student for a certain period",
+                response=inline_serializer(
+                    name="WorkedHoursAnalysisResponse",
+                    fields={
+                        "student_id": serializers.IntegerField(),
+                        "worked_minutes": serializers.IntegerField(),
+                        "student_on_tour_ids": serializers.ListField(child=serializers.IntegerField()),
+                    },
+                ),
+                examples=[
+                    OpenApiExample(
+                        "Successful Response 1",
+                        value={"student_id": 6, "worked_minutes": 112, "student_on_tour_ids": [1, 6, 9, 56, 57]},
```

Comment:
API response doesn't seem to be correct. I get a list of objects with postman instead of just 1 object.

Reply:
The list is correct. But I couldn't do that in the docs. Not sure why.